src/info/manifest.c

```diff
@@ -1663,42 +1663,81 @@ manifestBuildIncr(Manifest *this, const Manifest *manifestPrior, BackupType type
             }
         }
 
-        // Find files to reference in the prior manifest:
-        // 1) that don't need to be copied because delta is disabled and the size and timestamp match or size matches and is zero
-        // 2) where delta is enabled and size matches so checksum will be verified during backup and the file copied on mismatch
+        // Find files to (possibly) reference in the prior manifest
         bool delta = varBool(this->pub.data.backupOptionDelta);
 
         for (unsigned int fileIdx = 0; fileIdx < lstSize(this->pub.fileList); fileIdx++)
         {
             ManifestFile file = manifestFile(this, fileIdx);
 
-            // Check if prior file can be used
+            // Check if a prior file exists
             if (manifestFileExists(manifestPrior, file.name))
             {
                 const ManifestFile filePrior = manifestFileFind(manifestPrior, file.name);
 
-                if (file.copy &&
-                    ((filePrior.blockIncrMapSize > 0 && file.blockIncrSize > 0) ||
-                     (file.size == filePrior.size && (delta || file.size == 0 || file.timestamp == filePrior.timestamp))))
+                // If the file will be copied then preserve values from the prior file when needed. Some files may have been
+                // designated not to be copied at manifest build time, e.g. zero-length files when bundling.
+                if (file.copy)
                 {
-                    file.sizeRepo = filePrior.sizeRepo;
-                    file.checksumSha1 = filePrior.checksumSha1;
-                    file.checksumRepoSha1 = filePrior.checksumRepoSha1;
-                    file.reference = filePrior.reference != NULL ? filePrior.reference : manifestPrior->pub.data.backupLabel;
-                    file.checksumPage = filePrior.checksumPage;
-                    file.checksumPageError = filePrior.checksumPageError;
-                    file.checksumPageErrorList = filePrior.checksumPageErrorList;
-                    file.bundleId = filePrior.bundleId;
-                    file.bundleOffset = filePrior.bundleOffset;
-                    file.blockIncrSize = filePrior.blockIncrSize;
-                    file.blockIncrChecksumSize = filePrior.blockIncrChecksumSize;
-                    file.blockIncrMapSize = filePrior.blockIncrMapSize;
-
-                    // Perform delta if the file size is not zero
-                    file.delta = delta && file.size != 0;
-
-                    // Copy if the file has changed or delta
-                    file.copy = (file.size != 0 && file.timestamp != filePrior.timestamp) || file.delta;
+                    // If file size is equal to prior size then the file can be referenced instead of copied if it has not changed
+                    // (this must be determined during the backup).
+                    const bool fileEqual = file.size == filePrior.size;
```

Comment:
This would really be better named 'fileSizeEqual'...

Reply:
Agreed. Done in d40d79b4316.